tests/rptest/clients/kcl.py

```diff
@@ -273,35 +275,54 @@ def alter_partition_reassignments(self,
             reassignment_str += join_partitions
             cmd.append(reassignment_str)
 
-        lines = self._cmd(cmd).splitlines()
-        self._redpanda.logger.debug(lines)
-        ok_re = re.compile(
-            r"^(?P<topic>[a-z\-]+?) +(?P<partition>[0-9]+?) +OK$")
         no_broker_re = re.compile(
             r"^(?P<topic>[a-z\-]+?) +(?P<partition>[0-9]+?) +BROKER_NOT_AVAILABLE.*$"
         )
         bad_rep_factor_re = re.compile(
             r"^(?P<topic>[a-z\-]+?) +(?P<partition>[0-9]+?) +INVALID_REPLICATION_FACTOR.*$"
         )
-        for l in lines:
-            l = l.strip()
-            self._redpanda.logger.debug(l)
+        unknown_tp_re = re.compile(
+            r"^(?P<topic>[a-z\-]+?) +(?P<partition>[0-9]+?) +UNKNOWN_TOPIC_OR_PARTITION:.*$"
+        )
 
-            m = no_broker_re.match(l)
-            # No broker available means the partition did not find any eligible allocation nodes.
-            # See the map from cluster errors to kafka errors in kafka::map_topic_error_code()
-            if m is not None:
-                raise RuntimeError('No eligible allocation nodes')
+        lines = None
```

Comment:
is this necessary anymore

Reply:
Yes.
That is so we can return stdout and validate output. For example:
```py
def alter_partition_reassignments_with_kcl(
        kcl: KCL,
        topics: dict[str, dict[int, list[int]]],
        user_cred: Optional[dict[str, str]] = None,
        timeout_s: int = 10):
    ok_re = re.compile(r"^(?P<topic>[a-z\-]+?) +(?P<partition>[0-9]+?) +OK$")

    output = kcl.alter_partition_reassignments(topics,
                                               user_cred=user_cred,
                                               timeout_s=timeout_s)

    for line in output:
        assert ok_re.match(line.strip()) is not None

    return output
```

Plus, in case someone else uses kcl to AlterPartitionReassignments, they will also have the option to validate output.